pkg/server/plugin/datastore/sql/sql.go

```diff
@@ -88,11 +88,50 @@ type sqlDB struct {
 	raw              *sql.DB
 	*gorm.DB
 
+	stmtMu sync.RWMutex
+	stmts  map[string]*sql.Stmt
```

Comment:
Should this use a sync.Map (with a typed wrapper) instead of a global lock over the map?
It seems like it fits this use-case, and may avoid contention on a single lock.

I suspect it likely doesn't matter, though.

Reply:
I considered it but usually don't reach for sync.Map unless I have proven the codepath to be a hotspot because lockless programming is fraught with peril. However, I did some quick benchmarking and the sync.Map does lead to an order of magnitude less contention in this situation (e.g. 5microseconds v.s 25microseconds to take the lock with 500 concurrent query executors). The only downside is that it results in more prepared statements when the cache is cold since concurrent goroutines have to unconditionally prepare the statement between the failed `Load` and the hopeful `LoadOrStore`. That unnecessary load should only be a tiny spike right in the beginning, and is no worse than we have today. The tradeoff is probably worth the reduced contention. I'll switch over.

Reply:
The Load / LoadOrStore race seems fine to me; if it's ever a problem then I think we could pre-warm the cache in openDB.